### PR TITLE
Infinite recursion when trying to serialize Execution

### LINKED ISSUE #1130: Infinite recursion when trying to serialize `Execution`
```
ERROR c.s.s.b.c.RunExecutionController - Error during preprocessing. Reason: JSON encoding error: Infinite recursion (StackOverflowError); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Infinite recursion (StackOverflowError) (through reference chain: com.saveourtool.save.entities.User["originalLogins"]->org.hibernate.collection.internal.PersistentBag[0]->com.saveourtool.save.entities.OriginalLogin["user"]->com.saveourtool.save.entities.User["originalLogins"]->org.hibernate.collection.internal.PersistentBag[0]->com.saveourtool.save.entities.OriginalLogin["user"]->com.saveourtool.save.entities.User["originalLogins"]->org.hibernate.collection.internal.PersistentBag[0]->com.saveourtool.save.entities.OriginalLogin["user"]->com.saveourtool.save.entities.User["originalLogins"]->org.hibernate.collection.internal.PersistentBag[0]->com.saveourtool.save.entities.OriginalLogin["user"]->com.saveourtool.save.entities.User["originalLogins"]->org.hibernate.collection.internal.PersistentBag[0]->com.saveourtool.save.entities.OriginalLogin["user"]->com.saveourtool.save.entities.User["originalLogins"]...
at org.springframework.http.codec.json.AbstractJackson2Encoder.encodeValue(AbstractJackson2Encoder.java:233)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	*__checkpoint ? Request to POST http://orchestrator:5100/initializeAgents [DefaultWebClient]
Original Stack Trace:
		at org.springframework.http.codec.json.AbstractJackson2Encoder.encodeValue(AbstractJackson2Encoder.java:233)
		at org.springframework.http.codec.json.AbstractJackson2Encoder.lambda$encode$0(AbstractJackson2Encoder.java:150)
```

Occurs after #965, where `User.originalLogin` has been introduced.

### PR DESCRIPTION
**Pull request by Cheshiriks**
* fixed serialize and deserialize User
Closes #1130

### CODE REVIEW

File: save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        ).apply { id = 1 }
+
+        val originalLogin = OriginalLogin(
+            "admin",
+            user,
+            "basic"
+        ).apply { id = 4 }
+        user.apply { originalLogins = listOf(originalLogin) }
+
+        val result = objectMapper.writeValueAsString(user)
```
Comment (nulls) [inline save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt:61]: no need to write manually now -- you can put `user` to `bodyValue`

File: save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        webTestClient.post()
+            .uri("/internal/users/new")
+            .contentType(MediaType.APPLICATION_JSON)
+            .bodyValue(result)
+            .exchange()
+            .expectStatus()
+            .isOk
```
Comment (nulls) [inline save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt:69]: please add checking of result of this operation in database

File: api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt [hunk 3]
```diff
@@ -25,11 +23,7 @@ class StoringServerAuthenticationSuccessHandler(
 ) : ServerAuthenticationSuccessHandler {
     private val logger = LoggerFactory.getLogger(javaClass)
     private val objectMapper = ObjectMapper()
-        // This is necessary to remove the originalLogins from the User,
-        // otherwise we will get the kotlin.EmptyList class in Json and will not be able to serialize it.
-        .configure(SerializationFeature.WRITE_EMPTY_JSON_ARRAYS, false)
         .findAndRegisterModules()
-        .registerModule(CoreJackson2Module())
         .registerModule(OAuth2ClientJackson2Module())
```
Comment (petertrr) [inline api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt:27]: This is not needed either, moreover, if you look in their source code, they both call `SecurityJackson2Modules.enableDefaultTyping(context.getOwner());` which must be the cause of weird array serialization

Comment (nulls) [reply api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt:27]: maybe we don't need to initialize objectMapper and take it from spring context?

File: save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt]: Removed it

File: save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-backend/src/test/kotlin/com/saveourtool/save/backend/controller/UsersDetailsControllerTest.kt]: Added

File: api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt (hunk 3 above)
Comment (Cheshiriks) [reply api-gateway/src/main/kotlin/com/saveourtool/save/gateway/utils/StoringServerAuthenticationSuccessHandler.kt]: Fixed

**Review by nulls — APPROVED** [review]

**Review by petertrr — APPROVED** [review]

**Pull request merged**